array-reduce/main.js

```diff
@@ -0,0 +1,42 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((total, number) => total + number);
+
+console.log('sum:', sum);
+
+const product = numbers.reduce((product, number) => product * number);
+
+console.log('product:', product);
+
+const balance = account.reduce((total, transaction) => {
+  if (transaction.type === 'deposit') {
+    return total + transaction.amount;
+  } else {
+    return total - transaction.amount;
+  }
+}, 0);
+
+console.log('balance:', balance);
+
+const composite = traits.reduce((combinedObject, traits) => {
+  return Object.assign(combinedObject, traits);
+});
```

Comment:
Be sure to add an initial value (empty object) to copy all of the traits onto. otherwise your first trait is getting all others piled onto it.